app/src/androidTest/java/saveServiceTests/StartStopService.java

```diff
@@ -0,0 +1,133 @@
+package saveServiceTests;
+
+import android.content.Intent;
+
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+
+import com.shuttl.location_pings.config.open_lib.LocationsHelper;
+import com.shuttl.location_pings.service.LocationPingService;
+import com.shuttl.location_pings.service.LocationSaveService;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import testUtils.AssertUtils;
+import testUtils.BaseTestCase;
+import testUtils.LogUITest;
+import testUtils.UiUtils;
+import testUtils.customAnnotations.AutoTest_StartStopServices;
+
+@RunWith(AndroidJUnit4.class)
+public class StartStopService extends BaseTestCase {
+
+    @Before
+    public void setUp() {
+
+        UiUtils.stopSaveLocationServiceIfRunning(activityTestRule.getActivity().getApplication());
+        UiUtils.stopPingLocationServiceIfRunning(activityTestRule.getActivity().getApplication());
+    }
+
+
+    @AutoTest_StartStopServices
+    @Test
+    public void verifyStartStopLocationServicesViaInitModule() {
+
+        Intent intent = new Intent(appContext, LocationPingService.class);
+        intent.setAction("STOP");
+        // Start Both Services Via Init module .
+        LogUITest.debug("Starting 'Save Location Service' & 'Ping Location Service Via Init Module'");
+        LocationsHelper.INSTANCE.initLocationsModule(activityTestRule.getActivity().getApplication(), null, locationConfigs, locationPingServiceCallback, intent);
```

Comment:
we could make 3 helper methods 
- start ping service
- start save service
- start both

Reply:
Added ServiceHelper.java class which contains all helper methods for start / stop services.

Reply:
Each and every method of Service class has been tried and tested